### ISSUE TITLE
`no-test-import-export` false-positive with `@glint/type-test`

### ISSUE THREAD

**Issue by boris-petrov** [ember-cli/eslint-plugin-ember#1698]
Doing:

```ts
import { typeTest } from '@glint/type-test';
```

In a test file leads to an `ember/no-test-import-export` warning probably because of [this line](https://github.com/ember-cli/eslint-plugin-ember/blob/master/lib/rules/no-test-import-export.js#L50). However, [that's the way to do type-tests for templates](https://github.com/typed-ember/glint/tree/main/packages/type-test#loose-mode-templates) so there should be an exception.

**Comment by bmish** [comment]
The rule only intends to disallow importing from test file paths (ending in `-test`). It looks like we can just add an allowlist to the rule of common package names that are mistaken as test paths.
